Script to run tests on docker

torian opened the pull request:
Little script to run tests quickly and easily on docker by
excecuting:

```
 $> docker run --rm \
  -v ${PWD}:/opt/sharedbuffers \
  -w /opt/sharedbuffers \
  python:2.7 \
    /bin/sh run-tests.sh
```

Can also be used to run tests locally through virtualenv:

```
 $> sh ./run-tests.sh virtualenv
```

torian committed:
DEVOPS-1010: Script to run tests on docker

Little script to run tests quickly and easily on docker by
excecuting:

```
 $> docker run --rm \
  -v ${PWD}:/opt/sharedbuffers \
  -w /opt/sharedbuffers \
  python:2.7 \
    /bin/sh run-docker-tests.sh
```

sebamontini reviewed (approved):
LGTM

klaussfreire commented on README.rst:
```diff
@@ -156,3 +156,13 @@ rather than the first one, so multi-maps and simple mappings are binary compatib
+
+.. code::shell
+
+  $> docker run -v ${PWD}:/opt/sharedbuffers -w /opt/sharedbuffers python:2.7 /bin/sh run-docker-tests.sh
```
Docker is absolutely NOT the recommended way of running tests, so I wouldn't document this as the only way to do it.

If there's only one way to run tests documented, that's the way people will think is *the* way. This is not it.

Besides, that script there has nothing to do with docker. That's just a script to install requirements and run tests. You might as well drop the mention of docker and it would still be accurate.

torian replied:
Fair enough. I'll update the documentation.

klaussfreire commented on run-docker-tests.sh:
```diff
@@ -0,0 +1,12 @@
+
+set -e
+
+PATH=/usr/local/bin:$PATH
```
There's no reason/need to override the path

torian replied:
The idea is to run the python version that the image ships, which is located in `/usr/local/bin`, otherwise, it would run the tests with an older version (`/usr/bin`).

I can change that to an absolute path, if preferred.

klaussfreire replied:
It seems `/usr/local/bin` [is there by default](https://docs.docker.com/engine/reference/run/#env-environment-variables)

klaussfreire replied:
In any case, this script has no knowledge of the image being run, so any PATH manipulations should be done on the `docker run` command in the doc, that does know.

klaussfreire commented on run-docker-tests.sh:
```diff
@@ -0,0 +1,12 @@
+
+python --version
+
+pip install -q -r requirements.txt
```
Why quiet?

torian replied:
just to keep the interesting output, and not irrelevant things that pip might want to spit out.

klaussfreire replied:
The versions of installed requirements seem quite interesting to me

torian committed:
Fixed documentation on tests

torian committed:
Rename tests script

The script to run tests can be used locally or on docker.

By default, the script will run with your local python setup. You
can tell it otherwise running the script with the `virtualenv`
parameter:

```
  $> sh ./run-tests.sh virtualenv
```

It will create a virtualenv directory for you located in
`${VIRTUALENV_DIR}` which defaults to `venv`.

torian committed:
Removed virtualenv creation

klaussfreire commented on README.rst:
```diff
@@ -156,3 +156,20 @@ rather than the first one, so multi-maps and simple mappings are binary compatib
+Tests
+=====
+
+Running tests can be done locally or on docker, using the script `run-tests.sh`:
```
Include the virtualenv steps you had included earlier, that were a good idea.

```
Running tests can be done locally in a virtual env, or alternatively in a docker, using the script `run-tests.sh`:

.. code:: shell

  $> virtualenv venv
  $> . ./venv/bin/activate
  $> ./run-tests.sh
```

Make sure run-tests.sh has execute permission, btw.

klaussfreire replied:
After that, LGTM

torian committed:
Documentation
